src/table/boot.rs

```diff
@@ -203,6 +205,40 @@ impl BootServices {
         (self.free_pool)(addr).into()
     }
 
+    /// Stops execution until an event is signaled
+    ///
+    /// This function must be called at priority level TPL_APPLICATION. If an
+    /// attempt is made to call it at any other priority level, an `Unsupported`
+    /// error is returned.
+    ///
+    /// The input Event slice is repeatedly iterated from first to last until an
+    /// event is signaled or an error is detected. The following checks are
+    /// performed on each event:
+    ///
+    /// * If an event is of type NotifySignal, then an `InvalidParameter` error
+    ///   is returned.
+    /// * If an event is in the signaled state, the signaled state is cleared
+    ///   and the index of the event that was signaled is returned.
+    /// * If an event is not in the signaled state but does have a notification
+    ///   function, the notification function is queued at the event's
+    ///   notification task priority level. If the execution of the event's
+    ///   notification function causes the event to be signaled, then the
+    ///   signaled state is cleared and the index of the event that was signaled
+    ///   is returned.
+    ///
+    /// To wait for a specified time, a timer event must be included in the
+    /// Event slice.
+    ///
+    /// To check if an event is signaled without waiting, an already signaled
+    /// event can be used as the last event in the slice being checked, or the
+    /// check_event() interface may be used.
+    pub fn wait_for_event(&self, events: &mut [Event]) -> Result<usize> {
+        // FIXME: How to propagate the index of the faulty NotifySignal event?
```

Comment:
This functions provides extra information in the event of a certain error by writing to its OUT parameter. But since I'm using uefi-rs' general convention of hiding out parameters, the user won't be able to see it. Is it enough of a problem to justify deviating from the convention?

Reply:
How about returning a `core::result::Result<usize, (Status, usize)>`? It still provides the convenience of `Result` while not hiding the error index.

In fact, I though of doing so for the `Serial` protocol, since it would help with detecting timeout, but I though that in that case the user had the possibility of checking the returned `usize` against the buffer's full length.

Reply:
Thanks for the suggestion, implemented in the latest commit!

(One minor drawback is that one cannot transparently propagate results out of UEFI functions with "?" without using a type-erased Result type like failure's, but I think that's acceptable)